credential/integrity/signature_test.go

```diff
@@ -266,3 +266,200 @@ func getTestCredential() credential.VerifiableCredential {
+		resolver, err := resolution.NewResolver([]resolution.Resolver{key.Resolver{}}...)
+		assert.NoError(tt, err)
+
+		privKey, didKey, err := key.GenerateDIDKey(crypto.Ed25519)
+		assert.NoError(tt, err)
+		expanded, err := didKey.Expand()
+		assert.NoError(tt, err)
+		kid := expanded.VerificationMethod[0].ID
+		signer, err := jwx.NewJWXSigner(didKey.String(), kid, privKey)
+		assert.NoError(tt, err)
+
+		jwtPres := getTestJWTPresentation(tt, *signer, noCred)
+
+		verified, err := VerifyJWTPresentation(context.Background(), jwtPres, resolver)
+		assert.NoError(tt, err)
+		assert.True(tt, verified)
+	})
+
+	t.Run("valid presentation, bad credential", func(tt *testing.T) {
+		resolver, err := resolution.NewResolver([]resolution.Resolver{key.Resolver{}}...)
+		assert.NoError(tt, err)
+
+		privKey, didKey, err := key.GenerateDIDKey(crypto.Ed25519)
+		assert.NoError(tt, err)
+		expanded, err := didKey.Expand()
+		assert.NoError(tt, err)
+		kid := expanded.VerificationMethod[0].ID
+		signer, err := jwx.NewJWXSigner(didKey.String(), kid, privKey)
+		assert.NoError(tt, err)
+
+		jwtPres := getTestJWTPresentation(tt, *signer, badCred)
+
+		verified, err := VerifyJWTPresentation(context.Background(), jwtPres, resolver)
+		assert.Error(tt, err)
+		assert.Contains(tt, err.Error(), "verifying credential 0: parsing JWT: parsing credential token")
+		assert.False(tt, verified)
+	})
+
+	t.Run("valid presentation", func(tt *testing.T) {
+		resolver, err := resolution.NewResolver([]resolution.Resolver{key.Resolver{}}...)
+		assert.NoError(tt, err)
+
+		privKey, didKey, err := key.GenerateDIDKey(crypto.Ed25519)
+		assert.NoError(tt, err)
+		expanded, err := didKey.Expand()
+		assert.NoError(tt, err)
+		kid := expanded.VerificationMethod[0].ID
+		signer, err := jwx.NewJWXSigner(didKey.String(), kid, privKey)
+		assert.NoError(tt, err)
+
+		jwtPres := getTestJWTPresentation(tt, *signer, goodCred)
+		verified, err := VerifyJWTPresentation(context.Background(), jwtPres, resolver)
+		assert.NoError(tt, err)
+		assert.True(tt, verified)
+	})
+}
+
+type credStatus int
+
+const (
+	badCred credStatus = iota
+	goodCred
+	noCred
+)
+
+// cred status is a control flag. 0 = bad cred, 1 = good cred, 2 = no cred
+func getTestJWTPresentation(t *testing.T, signer jwx.Signer, credStatus credStatus) string {
+	cred := credential.VerifiableCredential{
+		ID:           uuid.NewString(),
+		Context:      []any{"https://www.w3.org/2018/credentials/v1"},
+		Type:         []string{"VerifiableCredential"},
+		Issuer:       signer.ID,
+		IssuanceDate: time.Now().Format(time.RFC3339),
+		CredentialSubject: map[string]any{
+			"id":            "did:example:123",
+			"favoriteColor": "green",
+			"favoriteFood":  "pizza",
+		},
+	}
+
+	signedCred, err := SignVerifiableCredentialJWT(signer, cred)
+	require.NoError(t, err)
+	require.NotEmpty(t, signedCred)
+
+	if credStatus == badCred {
+		// modify the signature to make it invalid
+		signedCred = signedCred[:len(signedCred)-25]
+	}
```

Comment:
Avoid conditionals to decrease cognitive effort on readers of the test. One way to do so is to separate this function into 3 functions `getTestJWTPresentationWithGoodCred`, `getTestJWTPresentationWithBadCred`, `getTestJWTPresentationWithoutCreds`.

Reply:
done